tests/callbacks/test_stochastic_weight_avg.py

```diff
@@ -217,3 +217,29 @@ def configure_optimizers(self):
         assert trainer.callbacks[0]._swa_lrs == (1e-3 if use_callbacks else 0.1)
     else:
         assert all(not isinstance(cb, StochasticWeightAveraging) for cb in trainer.callbacks)
+
+
+def test_swa_deepcopy(tmpdir):
+    """Test to ensure SWA Callback doesn't deepcopy dataloaders and datamodule potentially leading to OOM"""
+
+    class TestSWA(StochasticWeightAveraging):
+
+        def __init__(self, *args, **kwargs):
+            super().__init__(*args, **kwargs)
+            self.on_before_accelerator_backend_setup_called = False
+
+        def on_before_accelerator_backend_setup(self, trainer: 'Trainer', pl_module: 'LightningModule'):
+            super().on_before_accelerator_backend_setup(trainer, pl_module)
+            assert self._average_model.train_dataloader is not pl_module.train_dataloader
```

Comment:
What are you trying to test with this line?

Reply:
The pl_module train_dataloader has been wrapped into a PatchDataLoader at this stage. However, the train_dataloader on the average model should be the default function.